Bot/Discord/Handlers/InteractionHandler.cs

```diff
@@ -0,0 +1,126 @@
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogCritical($"Exception registering modules to interactionService: {ex.Message}", ex);
+                }
+
+            _interactionService.Log += LogMessage =>
+            {
+                if (LogMessage.Exception is CommandException cmdEx)
+                {
+                    _logger.Log(LogLevel.Error, cmdEx, $"Exception occured while executing Application Command. Message: {LogMessage.Message}");
+                }
+                else
+                {
+                    _logger.Log(LogLevel.Debug, $"Interaction service says: {LogMessage.Message}");
+                }
+                return Task.CompletedTask;
+            };
+
+            _client.InteractionCreated += HandleInteractionAsync;
+        }
+
+        // This feels sloppy af but if this doesnt happen (and there were changes to the commands) everything dies
+        public async Task RegisterCommandsAsync(bool ToGlobal = false)
+        {
+            try
+            {
+                if (ToGlobal)
+                {
+                    var res = await _interactionService.RegisterCommandsGloballyAsync();
+                    if (res != null)
+                    {
+                        _logger.LogInformation($"Successfully registered {res.Count} Application Commands globally");
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException();
+                    }
+                }
+                else
+                {
+                    var res = await _interactionService.RegisterCommandsToGuildAsync(_discordConfig.Value.GuildId);
+                    if (res != null)
+                    {
+                        _logger.LogInformation($"Successfully registered {res.Count} Application Commands to guild");
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException();
+                    }
+                }
+            }  
+            catch (Exception ex)
+            {
+                _logger.LogCritical("Failed to register Application Commands", ex);
+            }
+        }
+
+        private async Task HandleInteractionAsync(SocketInteraction interaction)
+        {
+            try
+            {
+                if (interaction == null || interaction.User.IsBot) return;
+
+                var ctx = new SocketInteractionContext(_client, interaction);
+                await _interactionService.ExecuteCommandAsync(ctx, _serviceProvider);
+            }
+            catch (Exception ex)
+            {
+                _logger.Log(LogLevel.Error, ex,
+                    $"Error handling interaction [User: {interaction.User} | Guild: {interaction.GuildId} | Channel: {interaction.Channel} | Id: {interaction.Id}]");
```

Comment:
Also, I suggest adding [SeriLog](https://serilog.net/) if you haven't and using their format (it's much nicer in my opinion). Format for this would be:

```cs
_logger.LogError("Error handling interaction [User: {User} | Guild: {Guild ...", interaction.User, interaction.Guild, ...);
```